src/util/strconv.h

```diff
@@ -127,4 +129,61 @@ static char *rm_strdupcase(const char *s, size_t len) {
   return ret;
 }
 
-#endif
\ No newline at end of file
+static char *rm_strdupcase_utf8(const char *s, size_t len) {
+  // TODO: set locale depending on the index language?
+  setlocale(LC_ALL, "en_US.UTF-8");
+
+  // Allocate memory for the destination string
+  char *ret = rm_strndup(s, len);
+  if (!ret) {
+    return NULL;
+  }
+
+  char *dst = ret;
+  char *src = dst;
+  mbstate_t state;
+  memset(&state, 0, sizeof(state));
+  wchar_t wc;
+  size_t len_wc;
+
+  while (*src) {
+    len_wc = mbrtowc(&wc, src, MB_CUR_MAX, &state);
+    if (len_wc == (size_t)-1 || len_wc == (size_t)-2) {
+      // Handle invalid multi-byte sequence
+      src++;
+      continue;
```

Comment:
What's invalid here?
It would be great to have coverage for this if it is easy to produce. If not - add some logging at least?
Let's understand when this will create odd behavior, and treat it accordingly (i.e., if this will never happen - ok, otherwise this needs better handling)

Reply:
I added a new tests, but the search for invalid sequences are reported as invalid syntax. What if we add an assert here also?

Reply:
An assertion sounds good to me, but let's try to understand when this should happen. If it shouldn't, an assertion is good.